packages/shared/src/store/dbHooks.ts

```diff
@@ -381,16 +377,27 @@ export const useMemberRoles = (chatId: string, userId: string) => {
 
 export const useGroupPreview = (groupId: string) => {
   const deps = useKeyFromQueryDeps(db.getGroup, groupId);
-  return useQuery({
+  const { data: group } = useGroup({ id: groupId });
+
+  const { data: groupPreview, ...rest } = useQuery({
     queryKey: [['groupPreview', groupId], deps],
     refetchOnReconnect: false,
     refetchOnMount: false,
-    enabled: !!groupId,
-    queryFn: async () => {
-      const [preview] = await syncGroupPreviews([groupId]);
-      return preview;
-    },
+    enabled: !!groupId && !group,
+    queryFn: async () => await api.getGroupPreview(groupId),
+    placeholderData: group ?? undefined,
   });
+
+  useEffect(() => {
+    if (groupId && !group && groupPreview) {
```

Comment:
it seems like if we did have a pre-existing group preview and it's stale, we wouldn't update the DB here?

might be fine, we'll probably scoop up the fresh metadata eventually. But thinking it could also make sense to call `syncGroupPreview` but make sure we only do it once per hook mount